src/openforms/registrations/contrib/objects_api/tests/test_templatetags.py

```diff
@@ -10,13 +10,15 @@
 
 
 class JsonSummeryTests(TransactionTestCase):
-    def test_render_json_summary(self):
-        form = FormFactory.create()
-        submission = SubmissionFactory.create(
-            form=form, completed=True, form_url="http://maykinmedia.nl/myform/"
+    def setUp(self):
+        self.form = FormFactory.create()
+        self.submission = SubmissionFactory.create(
+            form=self.form, completed=True, form_url="http://maykinmedia.nl/myform/"
         )
```

Comment:
Generally I'm not a fan of breaking this out into `setUp` methods, because it makes it harder to follow what is actually being set up if you look at the test code by itself.

I think this is also some poor factory usage in the original test, I'd think that this is better:

```py
form = FormFactory.create()
        step = SubmissionStepFactory.create(
            submission__form=form,
            submission__form_url="https://example.com/some-form",
            submission__completed=True,
            form_step__form=form,
            form_step__form_definition__slug="json-summery-step",
            form_step__form_definition__configuration={
                "display": "form",
                "components": [
                    {
                        "key": "voornaam",
                        "type": "textfield",
                        "registration": {
                            "attribute": RegistrationAttribute.initiator_voornamen,
                        },
                    },
                    {
                        "key": "achternaam",
                        "type": "textfield",
                        "registration": {
                            "attribute": RegistrationAttribute.initiator_geslachtsnaam,
                        },
                    },
                    {
                        "key": "tussenvoegsel",
                        "type": "textfield",
                        "registration": {
                            "attribute": RegistrationAttribute.initiator_tussenvoegsel,
                        },
                    },
                ],
            },
            data={
                "voornaam": "Foo",
                "achternaam": "Bar",
                "tussenvoegsel": "de",
            },
        )
```
and the lines below can use `step.submission` rather than `submission` or `self.submission`.

You can then do a similar thing in your own test, so you set up only the data that is actually needed and reading/debugging a test requires you to only look at the test itself.